src/Forms/GridField/GridField.php

```diff
@@ -444,6 +451,63 @@ private function initState(): void
         }
     }
 
+    /**
+     * Adds state for this gridfield from the request variables.
+     *
+     * If there is state already set on this GridField, that takes precedent
+     * over state from the request.
+     */
+    private function addStateFromRequest(): void
+    {
+        $request = $this->getRequest();
+        if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+        // if ($request->params()['Action']) {
```

Comment:
Guy, could you clarify why we should check 'Action', because this part prevent my changes work properly, so I commented this block out. 
If it has special purpose, I'll try to reconsider my changes.

Reply:
I don't know why I had that - if it's causing problems, and removing it fixes the problems, then I say just remove it.

Reply:
DONE